model_server_util_tools/model_packaging/export_model_utils.py

```diff
@@ -0,0 +1,239 @@
+        :return:
+        """
+        if export_file is None:
+            export_file = os.path.join(os.getcwd(), '{}{}'.format(model_name, MODEL_ARCHIVE_EXTENSION))
+
+        if os.path.exists(export_file):
+            raise ModelPackagingError(ModelPackagingErrorCodes.MODEL_ARCHIVE_ALREADY_PRESENT,
+                                      "model file {} already exists.".format(export_file))
+
+        return export_file
+
+    @staticmethod
+    def get_absolute_model_path(model_path):
+        """
+        Function to ge the absolute path, if given a relative path
+        :param model_path:
+        :return:
+        """
+        if model_path.startswith('~'):
+            return os.path.expanduser(model_path)
+        else:
+            return model_path
+
+    @staticmethod
+    def check_custom_model_types(model_path):
+        """
+        This functions checks whether any special handling is required for custom model extensions such as
+        .onnx, or in the future, for Tensorflow and PyTorch extensions.
+        :param model_path:
+        :return:
+        """
+        temp_files = [] # List of temp files added to handle custom models
+        files_to_exclude = [] # List of files to be excluded from .mar packaging.
+
+        files_set = set(os.listdir(model_path))
+        onnx_file = ModelExportUtils.find_unique(files_set, ONNX_TYPE)
+        if onnx_file is not None:
+            symbol_file, params_file = ModelExportUtils.convert_onnx_model(model_path, onnx_file)
+            files_to_exclude.append(onnx_file)
+            temp_files.append(os.path.join(model_path, symbol_file))
+            temp_files.append(os.path.join(model_path, params_file))
+
+        # More cases will go here as an if-else block
+
+        return temp_files, files_to_exclude
+
+    @staticmethod
+    def find_unique(files, suffix):
+        """
+        Function to find unique model params file
+        :param files:
+        :param suffix:
+        :return:
+        """
+        match = [f for f in files if f.endswith(suffix)]
+        count = len(match)
+
+        if count == 0:
+            return None
+        elif count == 1:
+            return match[0]
+        else:
+            params = {
+                '.params': ('.params', 'parameter file'),
+                '-symbol.json': ('...-symbol.json', 'symbol file'),
+                '.onnx': ('.onnx', 'ONNX model file'),
+            }[suffix]
+
+            message = 'mxnet-model-export expects only one ' + params[0] + ' file. Please supply the single ' + \
+                      params[1] + ' file you wish to export.'
+
+            raise ModelPackagingError(ModelPackagingErrorCodes.INVALID_MODEL_FILES, message)
+
+    @staticmethod
+    def convert_onnx_model(model_path, onnx_file):
+        """
+        Util to convert onnx model to MXNet model
+        :param model_path:
+        :param onnx_file:
+        :return:
+        """
+        from mxnet.contrib import onnx as onnx_mxnet
```

Comment:
We should fail gracefully if onnx or mxnet is not installed

Reply:
Done. Handled in : ab6a918fdd4745c8ccaf884409a7393163a7ddb9